src/Angular.js

```diff
@@ -941,6 +941,11 @@ function copy(source, destination, maxDepth) {
       for (key in source) {
         destination[key] = copyElement(source[key], maxDepth);
       }
+      // don't use Object.getOwnPropertyNames with RegExp to avoid copying RegExp lastIndex property
+    } else if (isObject(source) && typeof Object.getOwnPropertyNames === 'function' && !isRegExp(source)) {
```

Comment:
Not sure if it is reasonable to do `isRegExp` check here as I assume it can affect performance.

Not doing this check leads to copying own non-enumerable `lastIndex` property of `RegExp` object to destination.